tests/slurm_plugin/test_clustermgtd.py

```diff
@@ -2112,6 +2170,794 @@ def test_update_status(self, compute_fleet_status_manager, put_item_response, ex
+            ],
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", "launch_time"),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            {NodeType("queue2-c5xlarge", "queue1")},
+            {NodeType("queue1-c5xlarge", "queue1"), NodeType("queue1-c5large", "queue1")},
+            set(),
+            {NodeType("queue1-c5xlarge", "queue1"), NodeType("queue1-c5large", "queue1")},
+        ),
+        (
+            [
+                SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue1"),
+                SlurmNode("queue1-st-c5xlarge-2", "ip-2", "hostname", "some_state", "queue1"),
+            ],
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", "launch_time"),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            set(),
+            {NodeType("queue1-c5xlarge", "queue1"), NodeType("queue1-c5large", "queue1")},
+            set(),
+            {NodeType("queue1-c5xlarge", "queue1"), NodeType("queue1-c5large", "queue1")},
+        ),
+    ],
+)
+def test_handle_successfully_launched_nodes(
+    slurm_nodes,
+    private_ip_to_instance_map,
+    online_nodes_types,
+    bootstrap_failure_nodes_types,
+    successful_launched_partitions,
+    expected_bootstrap_failure_nodes_types,
+    mocker,
+):
+    # Test setup
+    mock_sync_config = SimpleNamespace(
+        terminate_max_batch_size=1,
+        launch_max_batch_size=5,
+        update_node_address=True,
+        region="us-east-2",
+        cluster_name="hit-test",
+        boto3_config=botocore.config.Config(),
+        dynamodb_table="table_name",
+        head_node_private_ip="master.ip",
+        head_node_hostname="master-hostname",
+        hosted_zone="hosted_zone",
+        dns_domain="dns.domain",
+        use_private_hostname=False,
+        instance_name_type_mapping={"c5xlarge": "c5.xlarge"},
+        protected_failure_count=10,
+    )
+    # Mock associated function
+    cluster_manager = ClusterManager(mock_sync_config)
+    mock_get_online_nodes_types = mocker.patch(
+        "slurm_plugin.clustermgtd.ClusterManager._get_online_nodes_types", return_value=online_nodes_types
+    )
+    cluster_manager._bootstrap_failure_nodes_types = bootstrap_failure_nodes_types
+    mock_reset_partition_failure_count = mocker.patch.object(
+        cluster_manager, "_reset_partition_failure_count", auto_spec=True
+    )
+    # Run test
+    cluster_manager._handle_successfully_launched_nodes(slurm_nodes, private_ip_to_instance_map)
+    # Assert calls
+    mock_get_online_nodes_types.assert_called_with(slurm_nodes, private_ip_to_instance_map)
+    assert_that(cluster_manager._bootstrap_failure_nodes_types).is_equal_to(expected_bootstrap_failure_nodes_types)
+    assert_that(mock_reset_partition_failure_count.call_count).is_equal_to(len(successful_launched_partitions))
```

Comment:
Can we also assert that `partition_failure_count` map is as expected after running the function?
Same comments applies to `test_handle_bootstrap_failure_nodes` and any function that will update  `partition_failure_count`.